packages/hash/frontend/src/components/layout/PageSidebar/AccountPageList/AccountPageList.tsx

```diff
@@ -21,25 +26,65 @@ type TreeElement = {
   children?: TreeElement[];
 };
 
-const renderTree = (node: TreeElement) => (
-  <PageTreeItem
-    hasChildren={node.children ? node.children.length > 1 : false}
-    key={node.entityId}
-    nodeId={node.entityId}
-    label={node.title}
-  >
-    {Array.isArray(node.children)
-      ? node.children.map((child) => renderTree(child))
-      : null}
-  </PageTreeItem>
-);
+const renderTree = (
+  node: TreeElement,
+  accountId: string,
+  depth: number = 0,
+) => {
+  return (
+    <PageTreeItem
+      key={node.entityId}
+      nodeId={node.entityId}
+      label={node.title}
+      depth={depth}
+      ContentProps={
+        {
+          /**
+           *  ContentProps type is currently limited to HtmlAttributes and unfortunately can't be augmented
+           *  Casting the type to any as a temporary workaround
+           * @see https://stackoverflow.com/a/69483286
+           * @see https://github.com/mui/material-ui/issues/28668
+           */
+          expandable: Boolean(
+            Array.isArray(node.children) ? node.children.length : node.children,
+          ),
+          url: `/${accountId}/${node.entityId}`,
+          depth,
+        } as any
+      }
+    >
+      {Array.isArray(node.children)
+        ? node.children.map((child) => renderTree(child, accountId, depth + 1))
+        : null}
+    </PageTreeItem>
+  );
+};
 
 export const AccountPageList: VoidFunctionComponent<AccountPageListProps> = ({
   currentPageEntityId,
   accountId,
 }) => {
   const { data } = useAccountPages(accountId);
   const router = useRouter();
+  const [loading, setLoading] = useState(false);
+
+  const { createUntitledPage } = useCreatePage(accountId);
+
+  // @todo handle loading/error states properly
+  const addPage = useCallback(async () => {
+    if (loading) {
+      return;
+    }
+
+    setLoading(true);
+    try {
+      await createUntitledPage();
+    } catch (err) {
+      // eslint-disable-next-line no-console -- TODO: consider using logger
+      console.error("Could not create page: ", err);
+      setLoading(false);
+    }
```

Comment:
**Blocking:** `setLoading(false)` only gets called when an error is thrown, it should be called in a `finally {}` block instead. However I don't think this is where this try catch block should live in the first place, it should instead be encapsulated in the `useCreatePage` hook.

Reply:
The comment you link doesn't justify why `setLoading(false)` is only called in the event of an error. Unless I'm missing something, I think this should be addressed.

Reply:
Sorry about that. I assumed I already fixed the first part. The comment was meant to be a reply to the second part (moving error handling to the hook).

Moved loading to the finally block here => [347db4a](https://github.com/hashintel/hash/pull/409/commits/347db4a21f392b75a44fdc779079f69e0ddf5a14)